lib/logstash/config/mixin.rb

```diff
@@ -201,23 +207,36 @@ def validate(params)
       return is_valid
     end # def validate
 
-    def validate_milestone
-      return true if @@milestone_notice_given
-      docmsg = "For more information about plugin milestones, see http://logstash.net/docs/#{LOGSTASH_VERSION}/plugin-milestones "
-      plugin_type = ancestors.find { |a| a.name =~ /::Base$/ }.config_name
-      case @milestone
-        when 0,1,2
-          @logger.warn(I18n.t("logstash.plugin.milestone.#{@milestone}", 
-                              :type => plugin_type, :name => @config_name,
+    def plugin_version
+      specification = Gem::Specification.find_by_name(plugin_gem_name)
+      major, minor, patch = specification.version.segments
+
+      Struct.new(:major, :minor, :patch)
+        .new(major, minor, patch)
+    end
+
+    def plugin_gem_name
+      [GEM_NAME_PREFIX, @plugin_type, @plugin_name].join('-')
+    end
+
+    def validate_plugin_version
+      return true if @@version_notice_given
+
+      if plugin_version.major < 1
```

Comment:
This validate method will always return true, right? Maybe it should be named something else and get pulled out of that validation chain, since it just deals with warnings.